### PR TITLE
Recursor: Add DNSSEC validation statistics

### LINKED ISSUE #3916: recursor: needs counters for DNSSEC status
i.e. one counter per DNSSEC state we determine (Secure, Insecure, Bogus, Indeterminate, Negative Trust Anchor).

### PR DESCRIPTION
**Pull request by pieterlexis**
Closes #3916

### CODE REVIEW

File: docs/markdown/recursor/stats.md [hunk 1]
```diff
@@ -26,6 +26,13 @@ The `rec_control get` command can be used to query the following statistics, eit
 * `client-parse-errors`: counts number of client packets that could not be parsed
 * `concurrent-queries`: shows the number of MThreads currently running
 * `dlg-only-drops`: number of records dropped because of delegation only setting
+* `dnssec-queries`: number of queries received with the DO and/or AD bit set
+* `dnssec-result-bogus`: number of DNSSEC validations that had the Bogus state
+* `dnssec-result-indeterminate`: number of DNSSEC validations that had the Indeterminate state
+* `dnssec-result-insecure`: number of DNSSEC validations that had the Insecure state
+* `dnssec-result-NTA`: number of DNSSEC validations that had the NTA (negative trust anchor) state
```
Comment (ahupowerdns) [inline docs/markdown/recursor/stats.md]: can you lowercase this? we have no policy on case sensitivity for metrics so for now I'd like to keep all of them lower case

File: pdns/validate-recursor.cc [hunk 2]
```diff
@@ -69,14 +74,19 @@ vState validateRecords(const vector<DNSRecord>& recs)
...
-  if(validrrsets.size() == cspmap.size()) // shortcut - everything was ok
+  if(validrrsets.size() == cspmap.size()) {// shortcut - everything was ok
+    g_stats.dnssecResults[Secure]++;
     return Secure;
+  }
 
   if(keys.empty()) {
+    g_stats.dnssecResults[Insecure]++;
     return Insecure;
   }
```
Comment (Habbie) [inline pdns/validate-recursor.cc]: I see like 6 instances of the ++ plus return - wonder if we can refactor that? a macro maybe?

**Comment by Habbie** [comment]
One nit, otherwise LGTM

File: pdns/validate-recursor.cc (hunk 2 above)
Comment (pieterlexis) [inline pdns/validate-recursor.cc]: Changed to an inline function

**Comment by pieterlexis** [comment]
could use another review

**Pull request merged**